Pinta.Effects/Effects/TwistEffect.cs

```diff
@@ -39,87 +40,96 @@ public override void LaunchConfiguration ()
-
-		twist = twist * twist * Math.Sign (twist);
-
-		int aaLevel = Data.Antialias;
-		int aaSamples = aaLevel * aaLevel + 1;
-		Span<PointD> aaPoints = stackalloc PointD[aaSamples];
-
-		for (int i = 0; i < aaSamples; ++i) {
-			PointD pt = new PointD (
-			    ((i * aaLevel) / (float) aaSamples),
-			    i / (float) aaSamples);
-
-			pt = pt with { X = pt.X - ((int) pt.X) };
-			aaPoints[i] = pt;
-		}
-
-		int width = src.Width;
 		ReadOnlySpan<ColorBgra> src_data = src.GetReadOnlyPixelData ();
 		Span<ColorBgra> dst_data = dst.GetPixelData ();
 
 		foreach (var rect in rois) {
 			for (int y = rect.Top; y <= rect.Bottom; y++) {
-				float j = y - hh;
-				var src_row = src_data.Slice (y * width, width);
-				var dst_row = dst_data.Slice (y * width, width);
-
+				float j = y - settings.HalfHeight;
+				var rowOffset = y * src.Width;
+				var src_row = src_data.Slice (rowOffset, src.Width);
+				var dst_row = dst_data.Slice (rowOffset, src.Width);
 				for (int x = rect.Left; x <= rect.Right; x++) {
-					float i = x - hw;
-
-					if (i * i + j * j > (maxrad + 1) * (maxrad + 1)) {
+					float i = x - settings.HalfWidth;
+					if (i * i + j * j > (settings.Maxrad + 1) * (settings.Maxrad + 1))
 						dst_row[x] = src_row[x];
-					} else {
-						int b = 0;
-						int g = 0;
-						int r = 0;
-						int a = 0;
-
-						for (int p = 0; p < aaSamples; ++p) {
-							float u = i + (float) aaPoints[p].X;
-							float v = j + (float) aaPoints[p].Y;
-							double rad = Math.Sqrt (u * u + v * v);
-							double theta = Math.Atan2 (v, u);
-
-							double t = 1 - rad / maxrad;
-
-							t = (t < 0) ? 0 : (t * t * t);
-
-							theta += (t * twist) / 100;
-
-							ref readonly ColorBgra sample = ref src.GetColorBgra (src_data, width,
-							    (int) (hw + (float) (rad * Math.Cos (theta))),
-							    (int) (hh + (float) (rad * Math.Sin (theta))));
-
-							b += sample.B;
-							g += sample.G;
-							r += sample.R;
-							a += sample.A;
-						}
-
-						dst_row[x] = ColorBgra.FromBgra (
-						    (byte) (b / aaSamples),
-						    (byte) (g / aaSamples),
-						    (byte) (r / aaSamples),
-						    (byte) (a / aaSamples));
-					}
+					else
+						dst_row[x] = GetFinalPixelColor (src, settings, src_data, j, i);
 				}
 			}
 		}
 	}
+
+	private sealed record MaterializedSettings (float HalfWidth, float HalfHeight, float Maxrad, float Twist, IReadOnlyList<PointD> AntialiasPoints);
+
+	private static ColorBgra GetFinalPixelColor (ImageSurface src, MaterializedSettings settings, ReadOnlySpan<ColorBgra> src_data, float j, float i)
+	{
+		int b = 0;
+		int g = 0;
+		int r = 0;
+		int a = 0;
+		int antialiasSamples = settings.AntialiasPoints.Count;
+		for (int p = 0; p < antialiasSamples; ++p) {
+			float u = i + (float) settings.AntialiasPoints[p].X;
+			float v = j + (float) settings.AntialiasPoints[p].Y;
+			double rad = Math.Sqrt (u * u + v * v);
+			double preTheta = Math.Atan2 (v, u);
+			double preT = 1 - rad / settings.Maxrad;
```

Comment:
I'm not a big fan of names like `preT` - they're not very meaningful so I'd rather see them replaced with mutable variables or given a meaningful name which explains what's happening

In this particular case I think you could eliminate `preTheta` entirely and just do `theta = Math.Atan2(v, u) + ((t * settings.Twist) / 100)`?

Reply:
@cameronwhite no problem, I just switched back to re-assignments

Reply:
As I understand it, you would be open to intermediate variables if they had a more meaningful name (i.e. instead of a name that means "the calculation I make before theta", we could give it the name of the mathematical entity it represents, or something with real-life meaning)?

Reply:
yeah exactly. In this case for example, the later variable from `t * t * t` maybe could have been called `t3` or `t_cubed` instead
But sometimes mutability can be more clear to read depending on the scenario